src/prototree/models.py

```diff
@@ -160,21 +160,27 @@ def predict(self, x: torch.Tensor) -> torch.Tensor:
         return torch.argmax(self.predict_probs(x), dim=-1)
 
 
+@dataclass
+class NodeSimilarity:
+    similarity: ImageProtoSimilarity
+    node: InternalNode
+
+
 @dataclass
 class LeafRationalization:
-    ancestor_similarities: list[ImageProtoSimilarity]
+    ancestor_sims: list[ImageProtoSimilarity]
     leaf: Leaf
 
     @property
     def proto_presents(self) -> list[bool]:
         """
```

Comment:
I don't think this should be a property. Apart from some runtime overhead, I think property docstrings are not shown to users by the IDE, and in this case the docstring is quite useful to know

Reply:
Good idea, I've changed this now.